### PR TITLE
Js to prevent advancing without form completion

### PR DESCRIPTION
**Pull request by facosta-eb**
"Next" button will prevent you from advancing if you didnt complete the form yet

### CODE REVIEW

File: static/js/taxpayer_creation.js [hunk 1]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
...
+// Moves to the next page of the form
+function next_page(esto){
+
 	if(animating) return false;
 	animating = true;
 
-	current_fs = $(this).parent();
-	next_fs = $(this).parent().next();
+	current_fs = esto.parent();
+	next_fs = esto.parent().next();
```
Comment (axelfernandez-eb) [inline static/js/taxpayer_creation.js:64]: 'Esto' should be in english?

Comment (facosta-eb) [reply static/js/taxpayer_creation.js:64]: "This" is a reserved word on javascript, so i used that

File: static/js/taxpayer_creation.js [hunk 2]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
 var animating; 
 
 
-$(".next").click(function(){
+// Checks if first part of form is complete
+function check_first_fieldset_complete(){
+	var token = true
+	$(".mandatory>input:text, .mandatory>input[type='number'], .mandatory>select, .mandatory>input[type='email']").each(function(index){
```
Comment (nicolasnunez-eb) [inline static/js/taxpayer_creation.js:10]: maybe we can first name this selector to something representative to know what it is and then execute the 'each' loop. i.e `let selector_name = $(".mandatory>input:text, .mandatory>input[type='number'], .mandatory>select, .mandatory>input[type='email']")`

File: static/js/taxpayer_creation.js [hunk 3]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
...
+		else if($(this).get(0).checkValidity() == false){
+			token = false;
+		}
+	});
+	return token
+}
+
+// Click on First "next" button
+$("#first-next").click(function(){
+	var esto = $(this);
```
Comment (nicolasnunez-eb) [inline static/js/taxpayer_creation.js:23]: Can we rename this to something like `current_component`

File: static/js/taxpayer_creation.js [hunk 4]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
...
+	}
+	else{
+		alert("Porfavor termine de completar el formulario antes de continuar");
+	}
+});
+
+// Checks if argentinian BANK form is filled
+function check_arg_bank_complete(){
+	var token = true;
+	$(".arg-bank>input:text, .arg-bank>input[type='number'], .arg-bank>select").each(function(index){
```
Comment (nicolasnunez-eb) [inline static/js/taxpayer_creation.js:35]: Same as the first comment.

File: static/js/taxpayer_creation.js [hunk 5]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
...
+		else if($(this).get(0).checkValidity() == false){
+			token = false;
+		}
+	});
+	return token
+}
+
+// Click on the arg-bank "next" button
+$("#arg-bank-next").click(function(){
+	var esto = $(this);
```
Comment (nicolasnunez-eb) [inline static/js/taxpayer_creation.js:48]: same as the second comment

File: static/js/taxpayer_creation.js [hunk 6]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
...
+	if (check_arg_bank_complete()){
+		next_page(esto);
+	}
+	else {
+		alert("Porfavor termine de completar el formulario antes de continuar");
+	}
+});
+
+// Moves to the next page of the form
+function next_page(esto){
```
Comment (nicolasnunez-eb) [inline static/js/taxpayer_creation.js:58]: Rename 'esto' to something like `component`

File: static/js/taxpayer_creation.js [hunk 7]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
...
+		}
+		else if($(this).get(0).checkValidity() == false){
+			token = false;
+		}
+	});
+	return token
+}
+
+// Click on First "next" button
+$("#first-next").click(function(){
```
Comment (nicolasnunez-eb) [inline static/js/taxpayer_creation.js:22]: Apparently this function follows the same logic as `$("#arg-bank-next").click` only changing the selector maybe we can think in a function that receives a selector and execute this logic.

File: static/js/taxpayer_creation.js [hunk 8]
```diff
@@ -4,12 +4,64 @@ var left, opacity, scale;
...
+	if (check_first_fieldset_complete()){
+		next_page(esto);
+	}
+	else{
+		alert("Porfavor termine de completar el formulario antes de continuar");
+	}
+});
+
+// Checks if argentinian BANK form is filled
+function check_arg_bank_complete(){
```
Comment (nicolasnunez-eb) [inline static/js/taxpayer_creation.js:33]: Same as one of the above comments this function follows the same logic as `check_first_fieldset_complete ` only changing the below selector maybe we can think in a function that receives a selector and execute this logic.

File: static/js/taxpayer_creation.js (hunk 7 above)
Comment (facosta-eb) [reply static/js/taxpayer_creation.js:22]: its the same logic but they call to diferent functions, one checks the completion of the first field set, and the other one checks the completion of Bank argentina fieldset

**Pull request merged**